### PR TITLE
feat(apikeys settings): add settings page for managing API keys

### PR DESCRIPTION
**Pull request by Camc314**
Adds settings page for adding/deleting/managing API keys

image

image

### CODE REVIEW

File: locales/en-US.json [hunk 1]
```diff
@@ -161,7 +162,23 @@
   "serverNotFound": "Server not found",
   "serverVersion": "Server version",
   "serverVersionTooLow": "Server version needs to be 10.7.0 or higher",
-  "settings": "Settings",
+  "settings": {
+    "settings": "Settings",
+    "apiKeys": {
+      "apiKeys": "Api Keys",
```
Comment (MrTimscampi) [inline locales/en-US.json:168]:
```suggestion
      "apiKeys": "API keys",
```

File: pages/settings/apikeys.vue [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      <v-btn v-if="apiKeys.length" color="error" @click="revokeAllApiKeys">
+        {{ $t('settings.apiKeys.revokeAll') }}
+      </v-btn>
+    </template>
+    <template #content>
+      <v-col>
+        <v-data-table :headers="headers" :items="apiKeys" class="elevation-1">
+          <!-- eslint-disable-next-line vue/valid-v-slot -->
+          <template #item.DateCreated="{ item }">
+            <span>
```
Comment (MrTimscampi) [inline pages/settings/apikeys.vue:16]:
```suggestion
            <span class="text-capitalize-first-letter">
```

Note: You'll need to move the class to one of the global stylesheets. It's already used in UserCard and logsAndActivity.

File: pages/settings/apikeys.vue [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                  $dateFns.parseJSON(item.DateCreated),
+                  new Date()
+                )
+              }}
+            </span>
+          </template>
+        </v-data-table>
+      </v-col>
+      <!-- Add API key dialog -->
+      <v-dialog
```
Comment (MrTimscampi) [inline pages/settings/apikeys.vue:28]: This should be moved to its own component.

File: pages/settings/apikeys.vue [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        :value="addingNewKey"
+        :width="width"
+        @click:outside="() => (addingNewKey = false)"
+      >
+        <v-col class="pa-0 add-key-dialog">
+          <v-card>
+            <v-card-title>Add Api Key</v-card-title>
```
Comment (MrTimscampi) [inline pages/settings/apikeys.vue:35]: This should be translated and should probably be `Add an API key` instead.

File: locales/en-US.json (hunk 1 above)
Comment (MrTimscampi) [reply locales/en-US.json:168]: This key also duplicates `settingsSections.apiKeys.name`, which we should probably avoid.

**Review by MrTimscampi — CHANGES_REQUESTED** [review]

File: locales/en-US.json (hunk 1 above)
Comment (Camc314) [reply locales/en-US.json]: I have removed the `settingsSection.apiKeys.name`.

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**